Tests/Unit/EmogrifierTest.php

```diff
@@ -609,6 +609,30 @@ public function selectorDataProvider()
                 => ['body:last-child {' . $styleRule . '} ', '#<p class="p-2">#'],
             'BODY:last-child matches last child'
                 => ['body:last-child {' . $styleRule . '} ', '#<p class="p-3" style="' . $styleRule . '">#'],
+            'wildcard selector [class*=P] matches first P element'    =>
+                ['[class*=p] {' . $styleRule . '} ', '#<p class="p-1" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P] matches second P element'    =>
+                ['[class*=p] {' . $styleRule . '} ', '#<p class="p-2" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P] matches third P element'    =>
+                ['[class*=p] {' . $styleRule . '} ', '#<p class="p-3" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-] matches first P elements'    =>
+                ['[class*=p-] {' . $styleRule . '} ', '#<p class="p-1" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-] matches second P elements'    =>
+                ['[class*=p-] {' . $styleRule . '} ', '#<p class="p-2" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-] matches second P elements'    =>
+                ['[class*=p-] {' . $styleRule . '} ', '#<p class="p-3" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-3] not matches first P elements'    =>
+                ['[class*=p-3] {' . $styleRule . '} ', '#<p class="p-1">#'],
+            'wildcard selector [class*=P-3] not matches second P elements'    =>
+                ['[class*=p-3] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-3] matches third P elements'    =>
+                ['[class*=p-3] {' . $styleRule . '} ', '#<p class="p-3" ' . $styleAttribute . '>#'],
+            'wildcard selector [class*=P-4] not matches first P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-4] not matches second P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-2">#'],
+            'wildcard selector [class*=P-4] not matches third P element'    =>
+                ['[class*=p-4] {' . $styleRule . '} ', '#<p class="p-3">#'],
```

Comment:
Generally, it's great that you've covered this with tests. I think we still can improve the tests.

For the wildcard selector, it's not important whether it's the first, second etc. P tag. Instead, I propose the following cases:

positive:
* selector value is an exact match with the attribute value
* selector value is a prefix of the attribute value
* selector value is a suffix of the the attribute value
* selector value is in the middle of the attribute value

negative cases:
* completely different strings
* non-empty selector value, empty attribute value
* non-empty selector value, missing attribute value
* empty selector value, non-empty attribute value (what's the expected behavior here? Is this even valid?)

What do you think?